src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/PerAlleleCollection.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect;
+
+import htsjdk.variant.variantcontext.Allele;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+
+/**
+ * A container for allele to value mapping.
+ *
+ * Each PerAlleleCollection may hold a value for each ALT allele and, optionally, a value for the REF allele.
+ * For example,
+ *
+ *   PerAlleleCollection<Double> alleleFractions = PerAlleleCollection.createPerAltAlleleCollection()
+ *
+ * may be a container for allele fractions for ALT alleles in a variant context. While
+ *
+ *   PerAlleleCollection<Double> alleleCount = PerAlleleCollection.createPerRefAndAltAlleleCollection()
+ *
+ * may hold the allele counts for the REF allele and all ALT alleles in a variant context.
+ *
+ *
+ **/
+public class PerAlleleCollection<X> {
```

Comment:
Request: could this class be in public? It does not contain any logic for Mutect, and I find that it could have quite interesting usages for other GATK-framework users....